blake3-hasher/src/lib.rs

```diff
@@ -76,7 +76,9 @@ mod tests {
         let val_hash = hash(val.as_bytes());
 
         let ext = "lM890t";
-        let ext_hash = hashv(&[&val_hash.to_bytes(), ext.as_bytes()]);
+        // Builds that enable copy on hash will trip this
+        #[allow(clippy::clone_on_copy)]
+        let ext_hash = hashv(&[&val_hash.clone().to_bytes(), ext.as_bytes()]);
```

Comment:
What do you think if we change the `to_bytes` implementation on the `Hash` type from:
```rust
pub const fn to_bytes(self) -> [u8; HASH_BYTES] {
    self.0
}
```
to:
```rust
pub const fn to_bytes(&self) -> [u8; HASH_BYTES] {
    self.0
}
```
The we can use `to_bytes` directly without the `clone_on_copy` issue. Since `Hash` is currently copy by default, I think this won't break things.